### ISSUE TITLE
ExtractCSS rule brakes HMR and SourceMaps in dev mode. CSS changes aren't registered

### ISSUE THREAD

**Issue by AndrewBogdanovTSS** [nuxt/nuxt.js#5390]
### Version

[v2.5.1](https://github.com/nuxt.js/releases/tag/v2.5.1)

### Reproduction link

[https://codesandbox.io/s/w08yyp72zk](https://codesandbox.io/s/w08yyp72zk)

### Steps to reproduce

1. Enable extractCss in nuxt.config.js
2. Run `nuxt` in dev mode
3. Try to update css/scss inside of the component

### What is expected ?

1. Page CSS is re-rendered dynamically without a need for a page refresh
2. Sourcemaps points to the correct file

### What is actually happening?

1. CSS changes appear only after full page refresh
2. CSS Sourcemaps point to generated css files - not on components it was initially written

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8936">#c8936</a>)</em></sub></div>

**Comment by manniL** [comment]
Should be probably reported to https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/ as well which is used for `extractCSS`

**Comment by mkurczewski** [comment]
Dirty fix:
```JS
{
  // ...
  extractCSS: process.env.NODE_ENV === 'production'
}
```

**Comment by manniL** [comment]
@AndrewBogdanovTSS 	As mentioned in https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/issues/161, does it still happen in nuxt-edge (which depends on v4.2.X)?

**Comment by levemsan** [comment]
The problem persists even with **nuxt-edge** and **extract-css-chunks-webpack-plugin^4.2.0**.

**Comment by AndrewBogdanovTSS** [comment]
@manniL yes, I tried to use latest version of **extract-css-chunks-webpack-plugin** and only after that logged an issue in their repo

**Comment by AndrewBogdanovTSS** [comment]
@manniL just tested it with latest Nuxt 2.6.0 beta which uses extract-css-chunks-webpack-plugin 4.2.0. and can confirm that it's still broken. I've also updated a repro link to use 4.2.0 explicitly

**Comment by gerbenqikker** [comment]
I also am experiencing this on Nuxt 2.6.0

**Comment by davision** [comment]
I can confirm this. It was super frustrating before I've realized, the extractCSS was the issue with HMR. The issue exists from v2.5.0 on.

**Comment by cannap** [comment]
Hi, i have this also in Nuxt.js v2.6.2

**Comment by ashnamuh** [comment]
I have the same issue on 2.7.1

**Comment by endcodedev** [comment]
I removed all extension from chrome (except vue extension) and it start reloading. Check if work in other browser, if so, then probably some extension block hot reloading

**Comment by aloiseau17** [comment]
There is any news about this issue ?
Related [issue on extract-css-chunks-webpack-plugin](https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/issues/161) is close but problem still occur for me.
